### DISCUSSION TITLE
Error migrating from nvim-lspconfig

### DISCUSSION THREAD

**Discussion by apmyplol** [mfussenegger/nvim-jdtls#579] [General]
As the title says, Im trying to migrate from `nvim-lspconfig` to `nvim-jdtls` because I want to try out debugging in nvim.

My Problem is that using the same `jdtls` config as in `nvim-lspconfig` yields the error `Client x quit with exit code 15 and signal 0`.
I could not find anything regarding this error.
What makes everything even weirder is, that if I try the minimal config provided in the readme, the LSP server starts, but if I replace
```lua
local config = {
    cmd = {'jdtls'},
    root_dir = vim.fs.dirname(vim.fs.find({'gradlew', '.git', 'mvnw'}, { upward = true })[1]),
}
require('jdtls').start_or_attach(config)
```

with

```lua
local project_name = vim.fn.fnamemodify(vim.fn.getcwd(), ":p:h:t")
local workspace_dir = "/path/to/workspace-root/" .. project_name

local config = {

    cmd = {

        "java", -- or '/path/to/java17_or_newer/bin/java'

        "-Declipse.application=org.eclipse.jdt.ls.core.id1",
        "-Dosgi.bundles.defaultStartLevel=4",
        "-Declipse.product=org.eclipse.jdt.ls.core.product",
        "-Dlog.protocol=true",
        "-Dlog.level=ALL",
        "-Xmx1g",
        "--add-modules=ALL-SYSTEM",
        "--add-opens",
        "java.base/java.util=ALL-UNNAMED",
        "--add-opens",
        "java.base/java.lang=ALL-UNNAMED",

        "-jar",
        "/usr/share/java/jdtls/plugins/org.eclipse.equinox.launcher_1.6.500.v20230717-2134.jar",

        "-configuration",
        "/usr/share/java/jdtls/config_linux",

   
        "-data",
        workspace_dir,
    },


    settings = {
        java = {},
    },

    init_options = {
        bundles = {},
    },
}

require("jdtls").start_or_attach(config)
```

As I said, this config works with `nvim-lspconfig` without any problems but does not with `nvim-jdtls`.
I will be very grateful if someone could point me into the right direction how I can fix this problem.

**Comment by mfussenegger** [comment]
Please check the [troubleshooting](https://github.com/mfussenegger/nvim-jdtls#troubleshooting) section

**Reply by apmyplol** [reply]
Yes I did and none of the options listed there worked for me.
What ended up working is changing the permissions of `/usr/share/java/jdtls/config_linux` which I find very weird.
Especially the fact that nvim-lspconfig worked with the given permissions and nvim-jdtls did not.

Either way fixed it, thanks for your help 🙏

**Comment by eruizc-dev** [comment]
One of the problems that JDTLS has is that the **config** directories must be writable by the language server. I'd assume you installed it from the AUR which gave me a lot of problems in the past.

They even try in the install script to make it writable, but it may not be working that well.
```
-# copy the configuration folder to tmp to be writable
-tmp_dir="$(mktemp -d /tmp/jdtls.XXXXX)"
-cp -R /usr/share/java/jdtls/config_linux "${tmp_dir}"
-# and ensure that it is removed on exit
-trap "{ rm -rf ${tmp_dir}; }" EXIT
```

You can either report that to the package maintainer or install it using tools like [Mason.nvim](https://github.com/williamboman/mason.nvim) which also bundles Lombok support
